src/RequestInformation.cs

```diff
@@ -203,11 +203,13 @@ private static object ReplaceEnumValueByStringRepresentation(object source)
         {
             var type = value.GetType();
 
-            if(Enum.GetName(type, value) is not { } name)
+            if (Enum.GetName(type, value) is not { } name)
                 throw new ArgumentException($"Invalid Enum value {value} for enum of type {type}");
 
-            if(type.GetField(name)?.GetCustomAttribute<EnumMemberAttribute>() is { } attribute)
+#pragma warning disable IL2075 // Enumerating Enum fields is always trimming/AOT safe - https://github.com/dotnet/runtime/issues/97737
+            if (type.GetField(name, BindingFlags.Static | BindingFlags.Public)?.GetCustomAttribute<EnumMemberAttribute>() is { } attribute)
                 return attribute.Value;
+#pragma warning restore IL2075
 
             return name.ToFirstCharacterLowerCase();
```

Comment:
Suppressing a warning this way only suppresses the Roslyn analyzer that runs during this library's build. The warning is still emitted for an end-user's app that uses this method.

Instead it should use the `[UnconditionalSuppressMessage]` attribute. Note that since it is best to only suppress the warning on the single call, we typically make a nested method to add the attribute to.

```suggestion
            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
                Justification = "Enumerating Enum fields is always trimming/AOT safe - https://github.com/dotnet/runtime/issues/97737")]
            static string? GetEnumMemberValue(Type enumType, string name) =>
                type.GetField(name, BindingFlags.Static | BindingFlags.Public)?.GetCustomAttribute<EnumMemberAttribute>() is { } attribute ? attribute.Value : null;

            return GetEnumMemberValue(type, name) ?? name.ToFirstCharacterLowerCase();
```

Reply:
Ah, TIL thanks, I'd seen `UnconditionalSuppressMessage` around but I wasn't aware of the functional difference.